packages/graph/hash_graph/lib/graph/src/store/pool.rs

```diff
@@ -6,7 +6,11 @@ use crate::store::Store;
 #[async_trait]
 pub trait StorePool: Send + Sync {
     type Error;
-    type Store: Store + Send;
+    type Store<'pool>: Store + Send + Sync
+    where
+        Self: 'pool;
```

Comment:
This using a GAT. It's possible to work around this by adding a new treat
```rust
trait HasStore<'pool> {
    type Store
}
```

and then bound `StorePool`:
```
pub trait StorePool: Send + Sync + for<'pool> HasStore<'pool> {
```

The return type of the functions would then change to
```rust
async fn acquire(&self) -> Result<<Self as HasStore<'_>>::Store, Self::Error>;
```

The behavior should be the same, but using GATs is a lot simpler hiere.

Reply:
I think it's fine to keep this, but perhaps unfamiliar to some. Maybe we should add a comment explaining the behavior?

Reply:
Hmm, I don't know as this is a language feature. I can still add a hint, that we use a GAT here if you want to.

Reply:
Discussed off-GitHub that it's okay not to document this language feature